pydsfapi/setup.py

```diff
@@ -1,18 +1,45 @@
 #!usr/bin/env python3
 
-from distutils.core import setup
+from setuptools import setup, find_packages
+import pathlib
 
-setup(name='pydsfapi',
-      version='3.2.0',
-      description='Python interface to access DuetSoftwareFramework',
-      author='Manuel Coenen',
-      author_email='manuel.coenen@gmail.com',
-      url='https://github.com/Duet3D/DSF-APIs',
-      packages=[
-          'pydsfapi',
-          'pydsfapi.commands',
-          'pydsfapi.initmessages',
-          'pydsfapi.model',
-          ],
-      license='LGPLv3',
-      )
+here = pathlib.Path(__file__).parent.resolve()
+long_description = (here / "README.md").read_text(encoding="utf-8")
+
+setup(
+    name="pydsfapi",
+    version="3.1.3",
+    description="Python interface to access DuetSoftwareFramework",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/Duet3D/DSF-APIs",
+    author="Manuel Coenen",
+    author_email="manuel.coenen@gmail.com",
+    classifiers=[
+        "Development Status :: 3 - Production/Stable",
+        "Intended Audience :: Developers",
+        "Topic :: Software Development :: Libraries",
+        "License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3 :: Only",
+    ],
+    keywords="Duet3D, DuetSoftwareFramework, DSF",
+    package_dir={"": "src"},
+    packages=find_packages(where="src"),
+    python_requires=">=3.6, <4",
```

Comment:
Last time I checked on DuetPi Python 3.5 was installed (I personally don't use DuetPi, so I do not check very often). So unless this has been updated to at least 3.6 this needs to be compatible to Python 3.5.

Reply:
https://duet3d.dozuki.com/Wiki/SBC_Setup_for_Duet_3 links to https://pkg.duet3d.com/DuetPi-lite.zip currently contains a `2021-02-15-DuetPi-lite.img` which ships a `python3.7` executable.

Raspbian Buster comes with python 3.7.
Raspbian Stretch comes with python 3.5 - but I don't think DuetDSF is "that old" to have ever used Stretch?

Reply:
Latest DuetPi seems to be shipped with 3.7. So everything fine.